packages/mgt-components/src/components/mgt-person/mgt-person.ts

```diff
@@ -1113,6 +1126,15 @@ export class MgtPerson extends MgtTemplatedComponent {
     let details = this.personDetailsInternal || this.personDetails || this.fallbackDetails;
 
     if (details) {
+      if ('personType' in details || 'userType' in details) {
+        details.entityType = 'user';
+      }
+      if ('initials' in details) {
+        details.entityType = 'contact';
+      }
+      if ('groupTypes' in details) {
+        details.entityType = 'group';
+      }
```

Comment:
This looks a lot like what our type guards could use instead of introducing the entityType property.